src/main/java/com/keshavarzi/airplanestats/model/UserEntity.java

```diff
@@ -0,0 +1,141 @@
+package com.keshavarzi.airplanestats.model;
+
+import jakarta.annotation.Nonnull;
+import jakarta.persistence.CascadeType;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.ForeignKey;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.JoinTable;
+import jakarta.persistence.ManyToMany;
+import jakarta.persistence.OneToMany;
+import jakarta.persistence.Table;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.Collection;
+import java.util.List;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@Entity
+@Table(name = "user", schema = "user_data")
+public class UserEntity implements Cloneable {
+
+    @Id
+    @Nonnull
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", table = "user", unique = true, nullable = false)
+    private Long userId;
+
+    @Nonnull
+    @Column(name = "email", table = "user", unique = true, nullable = false)
+    private String email;
+
+    @Nonnull
+    @Column(name = "password", table = "user", nullable = false)
+    private String password;
+
+    @Getter(AccessLevel.NONE)
+    @Setter(AccessLevel.NONE)
+    @OneToMany(mappedBy = "userEntity", targetEntity = RouteEntity.class, orphanRemoval = true, cascade = CascadeType.ALL)
+    private Collection<RouteEntity> savedRoutes;
+
+    // Exclusion is because associations impact performance
+    @Getter(AccessLevel.NONE)
+    @Setter(AccessLevel.NONE)
+    @OneToMany(mappedBy = "userEntity", targetEntity = Trip.class, orphanRemoval = true, cascade = CascadeType.ALL)
+    private Collection<Trip> savedTrips;
+
+    // Do not need to do it on the other entity, this table will be defined for both
+    @ManyToMany(fetch = FetchType.EAGER, targetEntity = RoleEntity.class)
+    @JoinTable(name = "user_role", schema = "user_data", joinColumns = @JoinColumn(name = "user_id", table = "user_data.role", referencedColumnName = "id", nullable = false,
+            foreignKey = @ForeignKey(name = "user_role_user_id_fk")),
+            inverseJoinColumns = @JoinColumn(name = "role_id", table = "role", referencedColumnName = "id", nullable = false,
+                    foreignKey = @ForeignKey(name = "user_role_role_id_fk")))
+    private Collection<RoleEntity> roleEntities;
+
+    public Collection<RoleEntity> getRoleEntities() throws CloneNotSupportedException {
+        return List.copyOf(this.roleEntities);
+    }
+
+    /**
+     * <p>A deep copy to set roleEntities field.
+     * Deep copy is to prevent unintentional/malicious changes through reference</p>
+     * @param roleEntities list of role entities
+     */
+    public void setRoleEntities(Collection<RoleEntity> roleEntities) {
+        this.roleEntities = List.copyOf(roleEntities);
+    }
```

Comment:
You should always have the input to a method be `final` to avoid future bugs. Also, either change the comment to reflect that this is NOT a deep copy or change the code to actually make a deep copy. A deep copy would mean that all objects within the list are recreated as well. In the current code you are just creating a *new* list that points to the original objects. (shallow copy)